osu.Game.Tests/Visual/Gameplay/TestSceneClicksPerSecond.cs

```diff
@@ -0,0 +1,307 @@
+
+            public MockFrameBasedClock()
+            {
+                Rate = 1;
+                IsRunning = true;
+            }
+
+            public void ProcessFrame()
+            {
+                CurrentTime += FRAME_INTERVAL * Rate * (Backwards ? -1 : 1);
+                TimeInfo = new FrameTimeInfo
+                {
+                    Current = CurrentTime,
+                    Elapsed = FRAME_INTERVAL * Rate * (Backwards ? -1 : 1)
+                };
+            }
+
+            public void Seek(double time)
+            {
+                TimeInfo = new FrameTimeInfo
+                {
+                    Elapsed = time - CurrentTime,
+                    Current = CurrentTime = time
+                };
+            }
+
+            public double ElapsedFrameTime => TimeInfo.Elapsed;
+            public double FramesPerSecond => 1 / FRAME_INTERVAL;
+            public FrameTimeInfo TimeInfo { get; private set; }
+        }
+
+        private class MockFrameStableClock : GameplayClock, IFrameStableClock
+        {
+            public MockFrameStableClock(MockFrameBasedClock underlyingClock)
+                : base(underlyingClock)
+            {
+            }
+
+            public void Seek(double time) => (UnderlyingClock as MockFrameBasedClock)?.Seek(time);
+
+            public IBindable<bool> IsCatchingUp => new Bindable<bool>();
+            public IBindable<bool> WaitingOnFrames => new Bindable<bool>();
+        }
+
+        private class MockDrawableRuleset : DrawableRuleset
+        {
+            public MockDrawableRuleset(Ruleset ruleset, IFrameStableClock clock)
+                : base(ruleset)
+            {
+                FrameStableClock = clock;
+            }
+
+#pragma warning disable CS0067
+            public override event Action<JudgementResult>? NewResult;
+            public override event Action<JudgementResult>? RevertResult;
+#pragma warning restore CS0067
+            public override Playfield? Playfield => null;
+            public override Container? Overlays => null;
+            public override Container? FrameStableComponents => null;
+            public override IFrameStableClock FrameStableClock { get; }
+
+            internal override bool FrameStablePlayback { get; set; }
+            public override IReadOnlyList<Mod> Mods => Array.Empty<Mod>();
+            public override IEnumerable<HitObject> Objects => Array.Empty<HitObject>();
+            public override double GameplayStartTime => 0;
+            public override GameplayCursorContainer? Cursor => null;
+
+            public override void SetReplayScore(Score replayScore)
+            {
+            }
+
+            public override void SetRecordTarget(Score score)
+            {
+            }
+
+            public override void RequestResume(Action continueResume)
+            {
+            }
+
+            public override void CancelResume()
+            {
+            }
+        }
+
+        #endregion
```

Comment:
I found `TestDrawableRuleset` in other tests so I might use and edit that. For the `IGameplayClock` I'm just going to use a `GameplayClockContainer` backed by a `ManualClock`. However I can't find an alternative for `IFrameStableClock` and since GameplayClock is no more I can't use my `MockFrameStableClock` as it was.